src/documents/IoCSS.md

```diff
@@ -47,74 +52,78 @@ fluid.uiOptions(".my-uio-container", {
+If the refactoring was even more thorough (involving wholesale removal of the target component, or a change in its important grades), the developer could still maintain stability of the external user API just by changing the `distributeOptions` block. 
+
+In terms of a standard discussion on [Design Patterns](https://en.wikipedia.org/wiki/Software_design_pattern "Design Patterns"), this use of `distributeOptions` could be seen as an automated and declarative scheme for achieving the ends of the [Facade Design Pattern](https://en.wikipedia.org/wiki/Facade_pattern "Facade Design Pattern"), without the need for either user or developer code.
+
+However, facade formation isn't the only possible function of an options distribution. They are also sufficiently powerful to encompass most uses of the so-called [Aspect-oriented programming](https://en.wikipedia.org/wiki/Aspect-oriented_programming) (AOP) by the
+use of the target root context `/` to match (and "advise", in AOP terminology) all components in a design matching a particular specification. Unlike traditional AOP techniques, however, these options distributions do not put ultimate and final power in the hands of the wielder - since the component holding
+the distribution itself can be either advised itself, or destroyed completely - thus "withdrawing" the "advice" from the system. 
 
-As well as converting the exposed options structure of a component into a more compact form, `distributeOptions` is also a powerful tool for maintaining API stability for a component or family of components. Since the binding of IoCSS selectors such as `that > templateLoader` onto child components is flexible, the component tree could be refactored in quite an aggressive way without requiring changes in either the user's configuration, or even the `distributeOptions` block itself. If the refactoring was even more thorough (involving wholesale removal of the target component, or a change in its important grades), the developer could still maintain stability of the external user API just by changing the `distributeOptions` block. In terms of a standard discussion on [Design Patterns](https://en.wikipedia.org/wiki/Software_design_pattern "Design Patterns"), the use of `distributeOptions` could be seen as an automated and declarative scheme for achieving the ends of the [Facade Design Pattern](https://en.wikipedia.org/wiki/Facade_pattern "Facade Design Pattern"), without the need for either user or developer code.
 
 ## distributeOptions format ##
 
-The `distributeOptions` option is a top-level block supported by every IoC-configured component, holding an array of objects (or single object) containing the following properties:
+The `distributeOptions` option is a top-level block supported by every IoC-configured component, holding an array of records, hash of records, or single record containing the following properties:
```

Comment:
Above, you changed "every IoC-configured component" to "every Infusion component" – should that change also be made here?